data/collector.go

```diff
@@ -285,10 +338,60 @@ func (c *collector) writeCaptureResults() {
 				c.logger.Error(errors.Wrap(err, fmt.Sprintf("failed to write to collector %s", c.target.Path())).Error())
 				return
 			}
+
+			c.maybeWriteToMongo(msg)
 		}
 	}
 }
 
+// TabularData is a denormalized sensor reading.
+type TabularData struct {
+	TimeRequested time.Time `bson:"time_requested"`
+	TimeReceived  time.Time `bson:"time_received"`
+	ComponentName string    `bson:"component_name"`
+	ComponentType string    `bson:"component_type"`
+	MethodName    string    `bson:"method_name"`
+	Data          bson.M    `bson:"data"`
+}
+
+func (c *collector) maybeWriteToMongo(msg *v1.SensorData) {
+	if c.mongoCollection == nil {
+		return
+	}
+
+	// currently vision.CaptureAllFromCamera and camera.GetImages are stored in .capture files as VERY LARGE
+	// tabular sensor data
+	// That is a mistake which we are rectifying but in the meantime we don't want data captured from those methods to be synced
+	// to mongo
+	if getDataType(c.methodName) == v1.DataType_DATA_TYPE_BINARY_SENSOR || c.methodName == captureAllFromCamera {
+		return
+	}
+
+	s := msg.GetStruct()
+	if s == nil {
+		return
+	}
+
+	data, err := PBStructToBSON(s)
+	if err != nil {
+		c.logger.Error(errors.Wrap(err, "failed to convert sensor data into bson"))
+		return
+	}
+
+	td := TabularData{
+		TimeRequested: msg.Metadata.TimeRequested.AsTime(),
+		TimeReceived:  msg.Metadata.TimeReceived.AsTime(),
+		ComponentName: c.componentName,
+		ComponentType: c.componentType,
+		MethodName:    c.methodName,
+		Data:          data,
+	}
+
+	if _, err := c.mongoCollection.InsertOne(c.cancelCtx, td); err != nil {
```

Comment:
Do we want to add a default timeout so this doesn't block if the database becomes unavailable, or is that on the user in your opinion?

Reply:
Very good call... I"ll add a timeout config option & default it to once a second.

Reply:
So... I don't think we need to do this. If the mongo url is invalid / if the client loses connection to the database, the client has a 2 second timeout built into it. 

I think this is fine for an initial implementation as blocking forever doesn't appear to be possible with this client.

Reply:
Oh cool! Agree that is sufficient, I was worried about the invalid URI case. Thanks!